extensions/arc/src/controller/generated/dusky/api/databaseRouterApi.ts

```diff
@@ -0,0 +1,2647 @@
+
+    public setApiKey(key: DatabaseRouterApiApiKeys, value: string) {
+        (this.authentications as any)[DatabaseRouterApiApiKeys[key]].apiKey = value;
+    }
+
+    set username(username: string) {
+        this.authentications.BasicAuth.username = username;
+    }
+
+    set password(password: string) {
+        this.authentications.BasicAuth.password = password;
+    }
+
+    set accessToken(accessToken: string | (() => string)) {
+        this.authentications.BearerAuth.accessToken = accessToken;
+    }
+
+    public addInterceptor(interceptor: Interceptor) {
+        this.interceptors.push(interceptor);
+    }
+
+    /**
+     *
+     * @summary Creates a database in a Dusky database service.
+     * @param ns The namespace of the database service.
+     * @param serviceName The name of the database service.
+     * @param duskyObjectModelsDatabase
+     */
+    public async createDuskyDatabase (ns: string, serviceName: string, duskyObjectModelsDatabase?: DuskyObjectModelsDatabase, options: {headers: {[name: string]: string}} = {headers: {}}) : Promise<{ response: http.IncomingMessage; body: DuskyObjectModelsDatabase;  }> {
+        const localVarPath = this.basePath + '/dusky/databases/{ns}/{serviceName}/databases'
+            .replace('{' + 'ns' + '}', encodeURIComponent(String(ns)))
+            .replace('{' + 'serviceName' + '}', encodeURIComponent(String(serviceName)));
+        let localVarQueryParameters: any = {};
+        let localVarHeaderParams: any = (<any>Object).assign({}, this._defaultHeaders);
+        const produces = ['application/json'];
+        // give precedence to 'application/json'
+        if (produces.indexOf('application/json') >= 0) {
+            localVarHeaderParams.Accept = 'application/json';
+        } else {
+            localVarHeaderParams.Accept = produces.join(',');
+        }
+        let localVarFormParams: any = {};
+
+        // verify required parameter 'ns' is not null or undefined
+        if (ns === null || ns === undefined) {
+            throw new Error('Required parameter ns was null or undefined when calling createDuskyDatabase.');
+        }
+
+        // verify required parameter 'serviceName' is not null or undefined
+        if (serviceName === null || serviceName === undefined) {
+            throw new Error('Required parameter serviceName was null or undefined when calling createDuskyDatabase.');
+        }
+
+        (<any>Object).assign(localVarHeaderParams, options.headers);
+
+        let localVarUseFormData = false;
+
+        let localVarRequestOptions: localVarRequest.Options = {
+            method: 'POST',
+            qs: localVarQueryParameters,
+            headers: localVarHeaderParams,
+            uri: localVarPath,
+            useQuerystring: this._useQuerystring,
+            json: true,
+            body: ObjectSerializer.serialize(duskyObjectModelsDatabase, "DuskyObjectModelsDatabase")
+        };
+
+        let authenticationPromise = Promise.resolve();
```

Comment:
This code style is also confusing

Reply:
FYI I'm not blocking on this, just noticed it.

Reply:
Yeah the code generated by openapi-generator isn't perfect.  But it's very convenient to generate all the routes and models from the controller's swagger spec.  openapi-generator was chosen over editor.swagger.io because editor.swagger.io doesn't support OpenAPI 3.x.